### ISSUE TITLE
R-Core Commands that are required to run a FULL GSE Analysis..

### ISSUE THREAD

**Issue by IsmailM** [GeoDiver/GEODiver#71]
Can you send me (@IsmailM) a script containing the R commands that are required to run a full GSE Analysis.

@Nazrath10R @sureshhewabi

**Comment by sureshhewabi** [comment]
Hi Ismail, did you mean script with testing variables?

**Comment by IsmailM** [comment]
Yep, it doesn't seem to work for me:

The files related with GSE3541 can be found here: ftp://ftp.ncbi.nlm.nih.gov//geo/series/GSE3nnn/GSE3541

```R
geodbpath <- '/Users/ismailm/.geodiver/DBs/GSE3541/GSE3541_series_matrix.txt.gz'

gset <- getGEO(filename = geodbpath, GSEMatrix = TRUE, AnnotGPL=TRUE)

if (length(gset) > 1) idx <- grep(gset@annotation, attr(gse, "names")) else idx <- 1
gset           <- gset[[idx]]
gpl            <- getGEO(annotation(gset))
featureData    <- gpl@dataTable@table
gene.names     <- featureData[, "Gene Symbol"]
organism       <- as.character(featureData[, "Species Scientific Name"][1])

X           <- exprs(eset) # Get Expression Data
pData       <- pData(eset)
rownames(X) <- gene.names

# KNN imputation
if (ncol(X) == 2) {
  X <- X[complete.cases(X), ] # KNN does not work when there are only 2 samples
} else {
  X <- X[rowSums(is.na(X)) != ncol(X), ] # remove rows with missing data
}

# Replace missing value with calculated KNN value
tryCatch({
  imputation <- impute.knn(X)
  X          <- imputation$data
}, error=function(e) {
  cat("ERROR: Bad dataset: Unable to run KNN imputation on the dataset.", file=stderr())
  quit(save = "no", status = 1, runLast = FALSE)
})

# If not log transformed, do the log2 transformed
if (scalable(X)) {
  X[which(X <= 0)] <- NaN # not possible to log transform negative numbers
  X <- log2(X)
}

if (! is.na(argv$outrdata)) {
  save(X, pData, gene.names, organism, file = argv$outrdata)
}
```

**Comment by Nazrath10R** [comment]
Aren't the { } brackets missing after the 
if (length(gset) > 1) statement
and after the 'else'

**Comment by IsmailM** [comment]
Nope, that if statement is perfect. You only need those brackets if there is more than one statement in the if block.

See those stackoverflow question for more information: http://stackoverflow.com/a/15587404/2619878

**Comment by IsmailM** [comment]
Fixed this part in 085077c
